go/vt/mysqlctl/xtrabackupengine.go

```diff
@@ -554,16 +557,28 @@ func (be *XtrabackupEngine) extractFiles(ctx context.Context, logger logutil.Log
 		// Create the decompressor if needed.
 		if compressed {
 			var decompressor io.ReadCloser
-
+			var deCompressionEngine = bm.CompressionEngine
+			if deCompressionEngine == "" {
+				// For backward compatibility. Incase if Manifest is from N-1 binary
+				// then we assign the default value of compressionEngine.
+				deCompressionEngine = PgzipCompressor
+			}
 			if *ExternalDecompressorCmd != "" {
-				decompressor, err = newExternalDecompressor(ctx, *ExternalDecompressorCmd, reader, logger)
+				if deCompressionEngine == ExternalCompressor {
+					deCompressionEngine = *ExternalDecompressorCmd
+					decompressor, err = newExternalDecompressor(ctx, deCompressionEngine, reader, logger)
+				} else {
+					decompressor, err = newBuiltinDecompressor(deCompressionEngine, reader, logger)
+				}
 			} else {
-				decompressor, err = newBuiltinDecompressorFromExtension(extension, *BuiltinDecompressor, reader, logger)
+				if deCompressionEngine == ExternalCompressor {
```

Comment:
If the manifest specifies `external` compression engine, but the user didn't specify `--external-decompressor`, then we return an error. Is that right?

Reply:
yes